pkg/util/winutil/iisconfig.go

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+//go:build windows
+// +build windows
+
+package winutil
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/fsnotify/fsnotify"
+)
+
+var (
+	// make global so that we can override for tests.
+	iisCfgPath = filepath.Join(os.Getenv("windir"), "System32", "inetsrv", "config", "applicationHost.config")
+)
+
+// DynamicIISConfig is an object that will watch the IIS configuration for
+// changes, and reload the configuration when it changes.  It provides additional
+// methods for getting specific configuration items
+type DynamicIISConfig struct {
+	watcher           *fsnotify.Watcher
+	path              string
+	wg                sync.WaitGroup
+	mux               sync.Mutex
+	stopChannel       chan bool
+	xmlcfg            *iisConfiguration
+	siteIDToNameAsTag map[uint32]string
+}
+
+func NewDynamicIISConfig() (*DynamicIISConfig, error) {
+	iiscfg := &DynamicIISConfig{
+		stopChannel: make(chan bool),
+	}
+	var err error
+
+	iiscfg.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// check for existence
+	_, err = os.Stat(iisCfgPath)
+	if os.IsNotExist(err) {
```

Comment:
Can we log an error here so it shows up in a flare and we can see that the path didn't exist at the expected location. Will help support cases since we know that is the reason.

Reply:
since the error is returned, it is already logged by the caller.